src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/template/SimpleTemplateEngineStepExecution.java

```diff
@@ -0,0 +1,107 @@
+import org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox;
+import org.jenkinsci.plugins.scriptsecurity.sandbox.Whitelist;
+import org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.ProxyWhitelist;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.io.IOException;
+
+import static org.apache.commons.lang.StringUtils.isBlank;
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+
+/**
+ * Execution of {@link SimpleTemplateEngineStep}.
+ *
+ * @author Martin d'Anjou
+ */
+public class SimpleTemplateEngineStepExecution extends AbstractFileOrTextStepExecution<Object> {
+    private static final long serialVersionUID = 1L;
+
+    private transient SimpleTemplateEngineStep step;
+
+    protected SimpleTemplateEngineStepExecution(@NonNull SimpleTemplateEngineStep step, @NonNull StepContext context) {
+        super(step, context);
+        this.step = step;
+    }
+
+    @Override
+    protected String doRun() throws Exception {
+        String fName = step.getDescriptor().getFunctionName();
+        if (isNotBlank(step.getFile()) && isNotBlank(step.getText())) {
+            throw new IllegalArgumentException(Messages.SimpleTemplateEngineStepExecution_tooManyArguments(fName));
+        }
+        
+        if (step.getBindings() == null) {
+            throw new IllegalArgumentException(Messages.SimpleTemplateEngineStepExecution_missingBindings(fName));
+        }
+
+        SimpleTemplateEngine engine = new SimpleTemplateEngine();
+        Template template = null;
+        if (isNotBlank(step.getFile())) {
+            FilePath f = ws.child(step.getFile());
+            if (f.exists() && !f.isDirectory()) {
+                try (InputStream is = f.read()) {
+                    template = engine.createTemplate(IOUtils.toString(is, StandardCharsets.UTF_8));
+                }
+            } else if (f.isDirectory()) {
+                throw new IllegalArgumentException(Messages.SimpleTemplateEngineStepExecution_fileIsDirectory(f.getRemote()));
+            } else if (!f.exists()) {
+                throw new FileNotFoundException(Messages.SimpleTemplateEngineStepExecution_fileNotFound(f.getRemote()));
+	    }
+        }
+        if (isNotBlank(step.getText())) {
+            template = engine.createTemplate(step.getText().trim());
+        }
+
+        String renderedTemplate = "";
+        final Template templateR = template;
+        final Map<String, Object> bindings = step.getBindings();
+        if (step.getRunInSandbox()) {
+            renderedTemplate = GroovySandbox.runInSandbox(
+                () -> {
+                    return templateR.make(bindings).toString();
+                },
+                new ProxyWhitelist(Whitelist.all())
+            );
+        } else {
+            renderedTemplate = templateR.make(bindings).toString();
```

Comment:
I believe this still allows for running unprotected code in the controller jvm, just set `runInSandbox` to `false` and you have bypassed the sandbox and now anyone that can edit a Jenkinsfile or configure a job effectively have administrator privileges. See https://javadoc.jenkins.io/plugin/script-security/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.html

Reply:
```suggestion
            ScriptApproval.get().using(script, GroovyLanguage.get());  //The code above needs to be reworked so that we can have the script string for this line.
            renderedTemplate = templateR.make(bindings).toString();
```